webapp/components/suggestion/at_mention_provider.jsx

```diff
@@ -126,22 +127,35 @@ function filterUsersByPrefix(users, prefix, limit, type) {
 }
 
 export default class AtMentionProvider {
-    handlePretextChanged(suggestionId, pretext) {
+    handlePretextChanged(suggestionId, pretext, argChannelId) {
```

Comment:
Instead of passing the channelId as an argument to handlePretextChanged, could you try passing it into the constructor for the AtMentionProvider when [it's created](https://github.com/mattermost/platform/blob/master/webapp/components/textbox.jsx#L39) instead? I'm not a fan of adding it to the SuggestionBox and handlePretextChanged since they're also used in places where there is no channel (like the search box).